### PR TITLE
Updated Kotlin to 1.6.21 with HMPP enabled

### PR DESCRIPTION
**Pull request by arkivanov**
This PR updates Kotlin to 1.6.21 and coroutines to 1.6.1. Since Kotlin 1.6.20, HMPP is enabled by default. So this PR is a follow-up to #638.

[KT-48811](https://youtrack.jetbrains.com/issue/KT-48811) is marked as fixed, however it's still relevant and is now being tracked in [KT-52193](https://youtrack.jetbrains.com/issue/KT-52193). Because of this, I had to supply default values to atomics in shared source sets, even though they have default values specified in `expect` common.

### CODE REVIEW

**Comment by arkivanov** [comment]
@CherryPerry Could you please run builds? I want to check if it compiles on CI. Thanks!

File: reaktive/src/jsMain/kotlin/com/badoo/reaktive/utils/SynchronizedCompat.kt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+package com.badoo.reaktive.utils
+
+internal actual inline fun <T> synchronizedCompat(lock: Any, block: () -> T): T = block()
```
Comment (CherryPerry) [inline reaktive/src/jsMain/kotlin/com/badoo/reaktive/utils/SynchronizedCompat.kt:3]: Should not `synchronized` be no-op in JS by default?
Or it was removed?

Comment (arkivanov) [reply reaktive/src/jsMain/kotlin/com/badoo/reaktive/utils/SynchronizedCompat.kt:3]: It was deprecated for a long time, and now doesn't compile in a shared source set. However it's still present in JS, and deprecated. Not sure what is the actual reason, but better to not use deprecated things anyway.

**Comment by arkivanov** [comment]
@CherryPerry This PR is ready for review, thanks!

File: reaktive/src/darwinCommonMain/kotlin/com/badoo/reaktive/scheduler/MainScheduler.kt [hunk 2]
```diff
@@ -59,7 +59,7 @@ internal class MainScheduler : Scheduler {
         }
 
         override fun cancel() {
-            operations.clear()
+            operations.clear(dispose = true)
```
Comment (CherryPerry) [inline reaktive/src/darwinCommonMain/kotlin/com/badoo/reaktive/scheduler/MainScheduler.kt:62]: `true` is a default value.

**Review by CherryPerry — APPROVED** [review]

File: reaktive/src/darwinCommonMain/kotlin/com/badoo/reaktive/scheduler/MainScheduler.kt (hunk 2 above)
Comment (arkivanov) [reply reaktive/src/darwinCommonMain/kotlin/com/badoo/reaktive/scheduler/MainScheduler.kt:62]: Yep, as I mentioned in the description, there is a bug and we must supply the value in shared source sets.

Comment (arkivanov) [reply reaktive/src/darwinCommonMain/kotlin/com/badoo/reaktive/scheduler/MainScheduler.kt:62]: Same for atomics

**Pull request merged**